be/src/storage/lake/compaction_policy.cpp

```diff
@@ -0,0 +1,159 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "storage/lake/compaction_policy.h"
+
+#include "common/config.h"
+#include "gutil/strings/join.h"
+#include "storage/lake/tablet.h"
+#include "storage/tablet_schema.h"
+
+namespace starrocks::lake {
+
+static const double kCompactionScoreThreshold = 0.999;
+
+class BaseAndCumulativeCompactionPolicy : public CompactionPolicy {
+public:
+    explicit BaseAndCumulativeCompactionPolicy(TabletPtr tablet) : _tablet(std::move(tablet)) {}
+    ~BaseAndCumulativeCompactionPolicy() override = default;
+
+    StatusOr<std::vector<RowsetPtr>> pick_rowsets(int64_t version) override;
+
+private:
+    double get_cumulative_score();
+    double get_base_score();
+    StatusOr<std::vector<RowsetPtr>> pick_cumulative_rowsets();
+    StatusOr<std::vector<RowsetPtr>> pick_base_rowsets();
+
+    TabletPtr _tablet;
```

Comment:
```suggestion
    Tablet _tablet;
```

Reply:
Tablet raw pointer is used in `Rowset` now, if use tablet copy, the tablet pointer in `Rowset` will become nullptr when the CompactionPolicy is destructed.
Let's optimize this later.